# Bonus: Temperature Analysis I

In [77]:
import pandas as pd
from datetime import datetime as dt

In [78]:
# "tobs" is "temperature observations"
input_temperature_df = pd.read_csv('Resources/hawaii_measurements.csv')
input_temperature_df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [79]:
# Convert the date column format from string to datetime
input_temperature_df['date'] = pd.to_datetime(input_temperature_df['date'], format='%Y-%m-%d')
input_temperature_df.head()

station       date  prcp  tobs
0  USC00519397 2010-01-01  0.08    65
1  USC00519397 2010-01-02  0.00    63
2  USC00519397 2010-01-03  0.00    74
3  USC00519397 2010-01-04  0.00    76
4  USC00519397 2010-01-06   NaN    73

In [80]:
# Set the date column as the DataFrame index
# Drop the date column
temperature_df = input_temperature_df.set_index('date')
temperature_df

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73
...                 ...   ...   ...
2017-08-19  USC00516128  0.09    71
2017-08-20  USC00516128   NaN    78
2017-08-21  USC00516128  0.56    76
2017-08-22  USC00516128  0.50    76
2017-08-23  USC00516128  0.45    76

[19550 rows x 3 columns]

### Compare June and December data across all years 

In [81]:
# Filter data for desired months
temperature_Jun_df = temperature_df.loc[(temperature_df.index.month==6)]
temperature_Dec_df = temperature_df.loc[(temperature_df.index.month==12)]

In [82]:
# Identify the average temperature for June
average_Jun = round(temperature_Jun_df["tobs"].mean(),2)
print(f'The average temperature for June is {average_Jun}.')

The average temperature for June is 74.94.


In [83]:
# Identify the average temperature for December
average_Dec = round(temperature_Dec_df["tobs"].mean(),2)
print(f'The average temperature for Dec is {average_Dec}.')

The average temperature for Dec is 71.04.


In [84]:
# Create collections of temperature data
temperature_Jun = temperature_Jun_df["tobs"].tolist()
temperature_Dec = temperature_Dec_df["tobs"].tolist()

In [85]:
# Run paired t-test
import scipy.stats as stats
stats.ttest_ind(temperature_Jun, temperature_Dec)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

__The null hypothesis (H0):__<br/>
There is no significant difference between the June's temperature and Dec's temperature.<br/>
__The alternative hypothesis (H1)__<br/>
There is a significant difference<br/>
Based on the p-value,pvalue=3.9025129038616655e-191, I got fron the calculation, there is less than a 3.9-e17% probability the null is correct.<br/>
Therefore, we reject the null hypothesis, and __accept the alternative hypothesis__.<br/>